crates/sui-faucet/src/faucet/simple_faucet.rs

```diff
@@ -919,6 +786,162 @@ impl Faucet for SimpleFaucet {
     }
 }
 
+pub async fn batch_gather(
+    request_consumer: &mut Receiver<(Uuid, SuiAddress, Vec<u64>)>,
+    requests: &mut Vec<(Uuid, SuiAddress, Vec<u64>)>,
+    batch_request_size: u64,
+) -> Result<(), FaucetError> {
+    let mut counter = 0;
+    // We take the first 500 items off the queue.
+    while counter < batch_request_size - 1 {
+        // We do need a timeout here or else it waits forever and gets stuck in this loop
```

Comment:
good catch! I just change the comment here. I think because the `BATCH_TIMEOUT` is so short and we would need to while loop with the tokio::select! we can get away with not implementing the exit signal here.